cobalt/browser/cobalt_browser_main_parts.cc

```diff
@@ -37,7 +37,21 @@
 
 namespace cobalt {
 
-CobaltBrowserMainParts::CobaltBrowserMainParts() = default;
+CobaltBrowserMainParts::CobaltBrowserMainParts(
+    std::unique_ptr<PrefService> experiment_config,
+    std::unique_ptr<PrefService> local_state,
+    std::unique_ptr<metrics_services_manager::MetricsServicesManager>
+        metrics_services_manager,
+    CobaltMetricsServicesManagerClient* metrics_services_manager_client)
+// : experiment_config_(std::move(experiment_config)),
+//   local_state_(std::move(local_state)),
+//   metrics_services_manager_(std::move(metrics_service_manager)),
+//   metrics_services_manager_client_(metrics_service_manager_client)
+{
+  global_features_ = std::make_unique<GlobalFeatures>(
+      std::move(experiment_config), std::move(local_state),
+      std::move(metrics_services_manager), metrics_services_manager_client);
+}
```

Comment:
nit: 
```
    : global_features_(new GlobalFeatures(
        std::move(experiment_config), std::move(local_state),
        std::move(metrics_services_manager), metrics_services_manager_client) {}
```

(prefer initialization over assignment in constructors, I think there's an entry in Scott Meyers' Effective C++ book).

Actually now that I see the uses of |global_features_|, does it need to be a unique_ptr<> ? FWICT a normal member variable would work just fine and would avoid fragmenting the heap; there's no code manipulating |global_features_| after construction. In general we tend to use unique_ptr<>s all the time and by default but for member variables, specially with move-only semantics, member vars are just fine and easier to read.

Reply:
Changed to member variable but is size of this variable not a concern? This GlobalFeatures class is supposed to be a replacement for BrowserProcessImpl + GlobalFeatures in Chrome. I'd expect this member variable to be very large